plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -303,11 +326,16 @@ export class GitlabOrgDiscoveryEntityProvider implements EntityProvider {
     }
 
     for (const group of groupResult) {
-      const entity = this.createGroupEntity(group, host);
+      const entity = this.groupTransformer(
+        group,
+        this.config,
+        this.groupNameTransformer,
+      );
 
       if (group.parent_id && idMapped.hasOwnProperty(group.parent_id)) {
-        entity.spec.parent = this.groupName(
-          idMapped[group.parent_id].full_path,
+        entity.spec.parent = this.groupNameTransformer(
+          idMapped[group.parent_id],
+          this.config,
```

Comment:
Hm. Now that we have a callback to form the entities, is it unfortunate to still have something that mutates the entity afterward? Should this thing be moved into the callback instead while we're at it? Thinking that maybe the `groupTransformer` could have a `parent?: GitLabGroup | undefined` argument or so. Then it could trivially do this work itself

Reply:
I fully understand what you don't like about the solution - I didn't like it either. The problem is that if I put the part into the callbackFunction, I also need to pass the list of all users.
The current processed group only knows the id of the parent group - to generate the default groupPath, I need the actual lookup of that group in the list of all groups.
So I end up not allowing to pass a groupTransformer but instead allow passing the groupEntitiesTransformer, that handles all groups and returns it's array - or what do you think?